packages/web-runtime/src/pages/resolvePublicLink.vue

```diff
@@ -69,18 +81,27 @@ import {
   queryItemAsString,
   useClientService,
   useRouteParam,
-  useRouteQuery
-} from 'web-pkg/src/composables'
+  useRouteQuery,
+  useStore, useUserContext
+} from "web-pkg/src/composables";
 import { useTask } from 'vue-concurrency'
 import { ref, unref, computed, defineComponent } from "@vue/composition-api";
+import isEmpty from "lodash-es/isEmpty";
 
 export default defineComponent({
   name: 'ResolvePublicLink',
   setup() {
     const { owncloudSdk } = useClientService()
+    const store = useStore()
     const token = useRouteParam('token')
+    const isUserContext = useUserContext({ store })
     const password = ref('')
-    const isPasswordRequiredTask = useTask(function* (signal) {
+    const tokenInfo = ref({})
+    const isPasswordRequiredTask = useTask(function* () {
+      if (!isEmpty(unref(tokenInfo))) {
+        return unref(tokenInfo).password_protected === 'true'
```

Comment:
Does the sdk really return an object with a property that contains true as a string? I don't even